dependencies.sh

```diff
@@ -27,10 +145,83 @@ else
 fi
 
 
+if cat /etc/os-release | grep "ubuntu" >> /dev/null; then
+
+        apt-get update
+
+        # Check if gcc is installed; if not, install it
+        if ! [ -x "$(command -v gcc)" ]; then
+        apt-get install -y gcc
+        fi
+
+        # Check if g++ is installed; if not, install it
+        if ! [ -x "$(command -v g++)" ]; then
+        apt-get install -y g++
+        fi
+
+        # Check if make is installed; if not, install it
+        if ! [ -x "$(command -v make)" ]; then
+        apt-get install -y make
+        fi
+
+elif cat /etc/os-release | grep -e "centos" >> /dev/null; then
+
+        yum update -y
+
+        # Check if gcc is installed; if not, install it
+        if ! [ -x "$(command -v gcc)" ]; then
+        yum install -y gcc
+        fi
+
+        # Check if g++ is installed; if not, install it
+        if ! [ -x "$(command -v g++)" ]; then
+        yum install -y gcc-c++
+        fi
+
+        # Check if make is installed; if not, install it
+        if ! [ -x "$(command -v make)" ]; then
+        yum install -y make
+        fi
+
+fi
+
+# install miniconda3
+if ! [ -x /usr/bin/miniconda3 ]
+then
+      wget https://repo.anaconda.com/miniconda/Miniconda3-py37_4.12.0-Linux-x86_64.sh \
+    && bash Miniconda3-py37_4.12.0-Linux-x86_64.sh -b -p /usr/bin/miniconda3/ \
+    && rm -f Miniconda3-py37_4.12.0-Linux-x86_64.sh
+else
+    echo "Found miniconda3. Continuing the installation...\n"
+fi
+
+if [ $? == 0 ] && [ -x /usr/bin/miniconda3 ]
+then
+        echo "miniconda3 installed successfully. Continuing the installation...\n"
+        if ! grep -q "/usr/bin/miniconda3/bin" /home/$(logname)/.bashrc || ! echo "$PATH" | grep -q "/usr/bin/miniconda3/bin"; then
+                echo "" >> /home/$(logname)/.bashrc
+                echo 'export PATH="/usr/bin/miniconda3/bin:$PATH"' >> /home/$(logname)/.bashrc
+                echo "miniconda3 added to PATH"
+        fi
+	export PATH=/usr/bin/miniconda3/bin:$PATH
+	conda update -y conda
+        if [ $? == 0 ];then conda install -c litex-hub --file conda_versions.txt -y ; else echo "Failed to update conda" ; exit ; fi
+        if [ $? == 0 ];then echo "Installed OpenROAD, Yosys, Skywater PDK, Magic and Netgen successfully" ; else echo "Failed to install conda packages" ; exit ; fi
+else
+	echo "Failed to install miniconda. Check above for error messages."
+	exit
+fi
+
+# install pip and/or download packages
 if which pip3 >> /dev/null
```

Comment:
Try hard-pointing to the pip3 binary from the miniconda3 location. Something like /usr/bin/miniconda3/bin/pip3.  Because of the conda version, this will not be executed.

Reply:
> Lets meet tomorrow so that I can show you how to test your updates at your side. Probably after the weekly monday meeting with Mehdi.

Sounds good. I can meet straight away after the meeting with Mehdi, at 12 PM EST.

Reply:
Could you elaborate this more? Since the script installs miniconda3 and adds it to the path environment variable before installing or checking for pip3, wouldn't /usr/bin/miniconda3/bin/pip3 already be part of the path by the time line 216 is executed? 

Also, what version of conda is required for successful execution?

Reply:
I agree. But just for the safe side, I think it's good to put that path.